syntax = "proto3";
import "google/protobuf/timestamp.proto";

message TaskData {
  enum Priority {
    NONE = 0;
    FIRST = 1;
    SECOND = 2;
    THIRD = 3;
  }
  string                     name        = 1;
  Priority                   prior       = 2;
  string                     label       = 3;
  google.protobuf.Timestamp  date        = 4;
  bool                       completed   = 5;

}

message TaskMessage {
  TaskData                   task = 1;
  repeated TaskMessage       subtasks = 8;
}

message TaskModelMessage {
  repeated TaskMessage tasks = 1;
}